### DISCUSSION TITLE
Best approach to deploy all flow versions for an agent

### DISCUSSION THREAD

**Discussion by jeff-slang** [xavidop/dialogflow-cx-cli#251] [Q&A] [answered]
I notice that cxcli does not support the [list flows API](https://cloud.google.com/dialogflow/cx/docs/reference/rest/v3/projects.locations.agents.flows/list). Curious if there is interest to expand the capabilities of cxcli to support this API?

I'm curious how people compose their cxcli deploy scripts without this capability. Is the current pattern to hardcode all flows to deploy in the shell script?

One alternative approach to leveraging the API is to consume the exported agent. For example, if one has the github export pulled to a directory, then you can parse the flow directory names:
` ls -l <root-directory>/flows`

This is far from ideal.

**Comment by xavidop** [comment]
Hi @jeff-slang ! great suggestion!

I have a question about the implementation since the cxcli uses the Display Name when it interacts with any resource. does this list command return the display names, the flow IDs or both in a JSON format?

I have to say that I did not add any list commands in any resource.

Thank you so much!
I am developing this in my free time. I hope you like the tool!
Xavi

**Reply by jeff-slang** [reply]
Hi Xavi,

Thanks for the quick response. I'm a big fan of the tool 😄 

The list command response returns a [flow object](https://cloud.google.com/dialogflow/cx/docs/reference/rest/v3/projects.locations.agents.flows#Flow) and this includes both of the attributes that you are asking about. The flow id is part of the `name` attribute, which would need to be parsed.

Field | Description
-- | --
name | stringThe unique identifier of the flow. Format: projects/<Project ID>/locations/<Location ID>/agents/<Agent ID>/flows/<Flow ID>.
displayName | stringRequired. The human-readable name of the flow.

**Reply by xavidop** [reply]
yeah, now that you mention the flow object, should I return the whole json object in an array of flows? or just the `displayName` and `name`?

**Reply by jeff-slang** [reply]
For my use-case, just the `displayName` and `name` are sufficient. I plan to pipe this into a flow version update command.

Having the rest of the data might be nice to have for other use-cases. Can this be easily parameterized in some sort of verbosity flag? Not sure if there is a pattern for this in the library. If not, maybe just the top-level name metadata attributes is enough.

**Reply by xavidop** [reply, marked as answer]
that makes sense! I can add a `--full` flag to show the entire data. btw if you check the PRs I am integrating data stores too
